### PR TITLE
cmd/reduce: use correct error detection; update license skipping

### PR DESCRIPTION
**Pull request by madelynnblue**
Error detection was incorrect and would panic, use a type instead of
value check.

Release note: None

### CODE REVIEW

File: pkg/cmd/reduce/main.go [hunk 1]
```diff
@@ -114,21 +114,23 @@ func reduceSQL(path, contains string, workers int, verbose bool) (string, error)
 	}
 
 	interesting := func(ctx context.Context, f reduce.File) bool {
-		cmd := exec.CommandContext(ctx, path, "demo", "--empty")
-		// Disable telemetry and license generation.
-		cmd.Env = []string{"COCKROACH_SKIP_ENABLING_DIAGNOSTIC_REPORTING", "true"}
```
Comment (knz) [inline pkg/cmd/reduce/main.go:119]: You probably need to keep the env var. `--disable-demo-license` is not sufficient to disable telemetry and sentry crash reporting.


(unless you intend to keep telemetry)

**Review by knz — APPROVED** [review]
LGTM with nit

File: pkg/cmd/reduce/main.go (hunk 1 above)
Comment (mjibson) [reply pkg/cmd/reduce/main.go:119]: TIL

**Pull request merged**